api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala

```diff
@@ -108,20 +114,32 @@ class Router(elasticClient: ElasticClient,
       val includes = params.include.getOrElse(V2WorksIncludes())
       worksService
         .findWorkById(id)(index)
-        .map {
-          case Right(Some(work: IdentifiedWork))           => workFound(work, includes)
-          case Right(Some(work: IdentifiedRedirectedWork)) => workRedirect(work)
-          case Right(Some(_))                              => workGone
-          case Right(None)                                 => workNotFound(id)
-          case Left(err)                                   => elasticError(err)
+        .flatMap {
+          case Right(Some(work: IdentifiedWork)) =>
+            val expandedPaths = params.expandPaths.getOrElse(Nil)
+            retrieveTree(index, work, expandedPaths).map {
+              workFound(work, _, includes)
+            }
```

Comment:
Would slightly prefer, for clarity:

```suggestion
            retrieveTree(index, work, expandedPaths).map { tree =>
              workFound(work, tree, includes)
            }
```

Also, can we flatMap here to avoid having to Future up all the other branches?